lib/Driver/ToolChain.cpp

```diff
@@ -116,9 +116,15 @@ std::unique_ptr<Job> ToolChain::constructJob(
 
   const char *responseFilePath = nullptr;
   const char *responseFileArg = nullptr;
-  if (invocationInfo.allowsResponseFiles &&
-      !llvm::sys::commandLineFitsWithinSystemLimits(
-          executablePath, invocationInfo.Arguments)) {
+
+  const bool forceResponseFiles =
+      C.getArgs().hasArg(options::OPT_driver_force_response_files);
+  assert((invocationInfo.allowsResponseFiles || !forceResponseFiles) &&
+         "Cannot force response file if platform does not allow it");
+
+  if (forceResponseFiles || (invocationInfo.allowsResponseFiles &&
```

Comment:
`invocationInfo.allowsResponseFiles` should still guard the whole thing, since one swiftc invocation can invoke multiple tools.

Reply:
Do you mean that if -driver-force-response-files is set on a platform that disallows them, you want the compiler just silently disregard the option? Or what do you want in that case? The current code is meant to fail an assertion.

Reply:
Thanks. Merged this a bit too quickly, let me know if you want a change. (I can't find the revert button, but can merge a fix.)